.github/workflows/node.js.yml

```diff
@@ -30,55 +34,79 @@ jobs:
         with:
           # npm cache files are stored in `~/.npm` on Linux/macOS
+          # Only needed if package-lock changed and we'll do npm ci
+          # It's possible we don't want to skip this since caches are evicted after 7 days of no access
+          # If no new packages are installed for a while, the install could take longer
+          # It might not be worth it though if the install only takes another minute vs 10-15s per action to cache this
           path: ~/.npm
-          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ matrix.node-version }}-${{ hashFiles('package-lock.json', 'packages/*/package-lock.json') }}
+          key: ${{ runner.os }}-npm-${{ hashFiles('package-lock.json') }}
           restore-keys: |
-            ${{ runner.os }}-build-${{ env.cache-name }}-${{ matrix.node-version }}
-            ${{ runner.os }}-build-${{ env.cache-name }}-
-            ${{ runner.os }}-build-
-            ${{ runner.os }}-
+            ${{ runner.os }}-npm-
 
-      - name: Cache linter caches
+      - name: Cache linters
         uses: actions/cache@v2
         with:
           path: |
             .eslintcache
             packages/*/.eslintcache
             .stylelintcache
             packages/*/.stylelintcache
-          key: ${{ runner.os }}-lintcache-${{ github.head_ref }}
+          key: ${{ runner.os }}-lintcache-${{ github.ref }}
           restore-keys: |
             ${{ runner.os }}-lintcache-
-            ${{ runner.os }}-
 
-      # Only need to install and bootstrap deps if package-locks changed
+      # Only need to install deps if package-lock changed
+      # Do this before restoring build cache since npm ci cleans node_modules
       - name: Install dependencies
         if: steps.cache-node-modules.outputs.cache-hit != 'true'
-        run: npm ci
+        run: npm ci --no-audit
+
+      # This is for webpack/terser cache mainly which helps build speed
+      # Cache on pushes to main should end up using the cache from the previous commit to main
+      # PRs should hit the cache on the base branch first, then future commits hit the cache on the PR
+      # Don't skip this on node_modules hit b/c package-lock will change less often than this cache
+      # The result would be less recent build caches if it were skipped
+      - name: Cache node_mocules/.cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            node_modules/.cache
+            packages/*/node_modules/.cache
+          key: ${{ runner.os }}-build-cache-${{ github.ref }}
+          restore-keys: |
+            ${{ runner.os }}-build-cache-
 
       - name: Build
         run: npm run build
 
       - name: Test (Pull Request)
         if: ${{ github.event_name == 'pull_request' }}
         run: |
-          npm run test:ci -- --changedSince origin/${{ github.event.pull_request.base.ref }}
+          npm run test:unit -- --cacheDirectory node_modules/.cache/jest
```

Comment:
Could use a comment here explaining why we run all the unit tests every time but only linting the main tests.